mutations.go

```diff
@@ -121,17 +121,36 @@ func (wh *mutationWH) applyMutationOnPod(pod corev1.Pod) ([]patchOperation, erro
 	}
 
 	if wh.imagePullSecret != "" {
-
-		op := "add"
-		if pod.Spec.ImagePullSecrets != nil {
-			op = "replace"
+		if pod.Spec.ImagePullSecrets == nil {
```

Comment:
It would be nice to have a small comment explaining the purposes of these branches (since there are now few possibilities depending on the input